core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -995,12 +995,7 @@ private void processWorkerAddedBlocks(MasterWorkerInfo workerInfo,
           Optional<BlockMeta> block = mBlockStore.getBlock(blockId);
           if (block.isPresent()) {
             workerInfo.addBlock(blockId);
-            BlockLocation blockLocation = BlockLocation.newBuilder()
-                .setWorkerId(workerInfo.getId())
-                .setTier(entry.getKey().getTier())
-                .setMediumType(entry.getKey().getMediumType())
-                .build();
-            mBlockStore.addLocation(blockId, blockLocation);
+            mBlockStore.addLocation(blockId, entry.getKey());
```

Comment:
Now that the `BlockLocation` contains the worker ID, I don't see a reason why we don't use that.

Reply:
This cut seems promising!
Can you also assert that workerIds match as expected?